### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by mcabbott**
This adds a link to the docs, and the logo, to the readme.

Matches https://github.com/FluxML/Optimisers.jl

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
+
 # NNlib.jl
 
+[![Documentation][docs-dev-img]][docs-dev-url]
 [![CI](https://github.com/FluxML/NNlib.jl/actions/workflows/ci.yml/badge.svg)](https://github.com/FluxML/NNlib.jl/actions/workflows/ci.yml)
 [![Coverage](https://codecov.io/gh/FluxML/NNlib.jl/branch/master/graph/badge.svg)](https://codecov.io/gh/FluxML/NNlib.jl) 
-[![Documentation](https://img.shields.io/badge/docs-fluxml.ai-blue)](https://fluxml.ai/Flux.jl/stable/models/nnlib/)
 
+[docs-stable-img]: https://img.shields.io/badge/docs-stable-blue.svg
+[docs-stable-url]: https://fluxml.ai/NNlib.jl/stable/
```
Comment (ToucheSir) [inline README.md:10]: Will this work once we tag a new version?

Comment (mcabbott) [reply README.md:10]: Oh I don't know, I copied from Optimisers where it also doesn't work.

Comment (ToucheSir) [reply README.md:10]: Worth a try then I guess.

**Review by ToucheSir — APPROVED** [review]

**Comment by Saransh-cpp** [comment]
Documentation for `v0.8.10` was not generated, see - https://github.com/FluxML/NNlib.jl/actions/runs/3314804815/jobs/5474567079. Looks like there was a problem with `DOCUMENTER_KEY`.

Given that the tag exists now, the documentation workflow can be (re)triggered using -
> **Note**
> I am not sure if this would work without resolving the `DOCUMENTER_KEY` issue. Someone with push access to `NNlib` could try this!
```
git tag -a v0.8.10+docs -m "Build and deploy docs for v0.8.10 manually"
git push --tags
```

**Pull request merged**